tests/unittests/test_handler/test_schema.py

```diff
@@ -447,4 +447,21 @@ def test_all_integration_test_cloud_config_schema(self):
         if errors:
             raise AssertionError(', '.join(errors))
 
+
+def _get_schema_doc_examples():
+    examples_dir = Path(
+        cloudinit.__file__).parent.parent / 'doc' / 'examples'
+    assert examples_dir.is_dir()
+
+    all_text_files = (f for f in examples_dir.glob('cloud-config*.txt')
+                      if not f.name.startswith('cloud-config-archive'))
+    return all_text_files
+
+
+class TestSchemaDocExamples:
+    @pytest.mark.parametrize("example_path", _get_schema_doc_examples())
+    @skipUnlessJsonSchema()
+    def test_schema_doc_examples(self, example_path):
+        validate_cloudconfig_file(str(example_path), get_schema())
```

Comment:
That's great :)

I just have a suggestion on the `get_schema` call, This functions does some processing to bundle together all of the config modules schemas. Maybe we could store its result into a parameter and reuse it for all of the `validate_cloudconfig_file` calls

Reply:
Okay, my comment is no longer needed hahaha